src/settings.rs

```diff
@@ -54,25 +54,27 @@ impl Settings {
     pub fn saving_location(&self) -> PathBuf {
         let stored_saving_location: PathBuf = self.0.get("saving-location");
 
-        if !stored_saving_location.as_os_str().is_empty() {
-            return stored_saving_location;
-        }
-
         let saving_location =
             glib::user_special_dir(glib::UserDirectory::Videos).unwrap_or_else(glib::home_dir);
 
+        if !stored_saving_location.as_os_str().is_empty() {
+            return self.create_saving_location(stored_saving_location);
+        }
+
         let kooha_saving_location = saving_location.join("Kooha");
+        self.create_saving_location(kooha_saving_location)
+    }
 
-        if let Err(err) = fs::create_dir_all(&kooha_saving_location) {
+    pub fn create_saving_location(&self, saving_location: PathBuf) -> PathBuf {
```

Comment:
I would put it as a standalone function, instead of an associated function, since &self is not accessed.

Also, I would name it `ensure_location`, and make it private since it is not accessed outside the module.

Reply:
Thank you very much! Merging it